src/odemis/gui/win/acquisition.py

```diff
@@ -1314,6 +1321,101 @@ def on_acquisition_done(self, future):
+            self.vp_correlation_tl.canvas.cryotarget_fm_overlay)
+        self.vp_correlation_tr.canvas.cryotarget_fib_overlay.active.value = True
+        self.vp_correlation_tr.canvas.add_world_overlay(
+            self.vp_correlation_tr.canvas.cryotarget_fib_overlay)
+
+        # stream bar controller
+        self.streambar_controller = StreamBarController(
+            tab_data,
+            self.pnl_correlation_streams,
+            static=True,
+        )
+
+        # correlation points controller
+        self._correlation_points_controller = CorrelationPointsController(self)
+
+        # Toolbar
+        self.tb = self.correlation_toolbar
+        for t in guimod.TOOL_ORDER:
+            if t in tab_data.tool.choices:
+                self.tb.add_tool(t, tab_data.tool)
+        # Add fit view to content to toolbar
+        self.tb.add_tool(TOOL_ACT_ZOOM_FIT, self.view_controller.fitViewToContent)
+        self.btn_cancel.Bind(wx.EVT_BUTTON, self.on_cancel)
+        self.Bind(wx.EVT_CLOSE, self.on_cancel)
+
+    @property
+    def correlation_points_controller(self):
+        return self._correlation_points_controller
+
+    def _create_views(self, viewports):
+        """
+        Create views depending on the actual hardware present
+        return OrderedDict: as needed for the ViewPortController
+        """
+        vpv = collections.OrderedDict([
+            (viewports[0], # focused view
+             {"name": "FLM view",
+              "stream_classes": OpticalStream,
+              }),
+            (viewports[1],
+             {"name": "FIB view",
+              "stream_classes": EMStream,
+              }),])
+        return vpv
+
+    def remove_all_streams(self):
+        """
+        Remove the streams we added to the view on creation
+        Must be called in the main GUI thread
+        """
+        # Ensure we don't update the view after the window is destroyed
+        self.streambar_controller.clear()
+        gc.collect()  # To help reclaiming some memory
+
+    def terminate_listeners(self):
+        """
+        Disconnect all the connections to the streams.
+        Must be called in the main GUI thread.
+        """
+        # stop listening to events
+        self._view.stream_tree.flat.unsubscribe(self.on_streams_changed)
+        self.remove_all_streams()
+        # Set the streambar controller to None so it wouldn't be a listener to stream.remove
+        self.streambar_controller = None
+
+    def on_close(self, evt):
+        """Close event handler that executes various cleanup actions"""
+        assert self._correlation_points_controller.is_processing == False
+
+        self.EndModal(wx.ID_CANCEL)
+
+    def on_cancel(self, evt):
+        """Handle acquisition cancel button click"""
+        logging.info("Close button clicked, exiting multipoint correlation window")
+        while self._correlation_points_controller.is_processing:
+            continue
```

Comment:
That is very CPU intensive, which is a pity as you're waiting for a CPU-intensive task to finish. You should at least add a `sleep()`. Better is to change `is_processing` to a `threading.Event` (named for instance `processing_complete`), so that you can just wait for it with `processing_complete.wait()`.

Reply:
Added the time.sleep(0.1)